elastic/kibana
# [Response Ops][Task Manager] Return `HealthStatus.Warning` instead of error when task manager stats are not yet available.

Resolves https://github.com/elastic/kibana/issues/168237

## Summary

This PR fixes a bug in the task manager status calculations that would return `0` for the "cold timestamp" when it didn't exist inside the stats and compare the difference of `now - 0` against the required freshness threshold. This was causing the task manager status to initially be set to `Error` on startup (when there were no task manager stats available at all). Updated to return a warning with an appropriate reason message.

## To verify:
Run ES & Kibana on `main`. See startup logs warning about task manager health:
```
[2023-10-25T15:27:00.740-04:00][WARN ][status.plugins.taskManager] taskManager plugin is now degraded: Task Manager is unhealthy - Reason: setting HealthStatus.Error because of expired cold timestamps
[2023-10-25T15:27:00.860-04:00][WARN ][status] Kibana is now degraded: 0 service(s) and 1 plugin(s) are degraded: taskManager
```

Switch to this branch and run ES & Kibana. There should be no warnings about task manager errors in the logs.
